### PR TITLE
NET-340

### PR DESCRIPTION
**Pull request by uGiFarukh**
## Describe your changes
* Added warning message when user tries to join another network where it is already joined with an existing network on another server.

## Provide testing steps
-> On a fresh netclient instance, join network1 of server1 with netclient join command and enrollment key.
-> Join another network2 of another server2 with netclient join command and another enrollment key.
-> This should produce the warning log message.

## Checklist before requesting a review
- [x] My changes affect only 10 files or less.
- [x] I have performed a self-review of my code and tested it.
- [ ] If it is a new feature, I have added thorough tests, my code is <= 1450 lines.
- [ ] If it is a bugfix, my code is <= 200 lines.
- [ ] My functions are <= 80 lines.
- [x] I have had my code reviewed by a peer.
- [ ] My unit tests pass locally.
- [x] Netclient & Netmaker are awesome.

### CODE REVIEW

**Review by theguy951357 — APPROVED** [review]
tested and passed

File: cmd/join.go [hunk 1]
```diff
@@ -29,6 +30,9 @@ user: netclient join -s <server> -u <user_name> // attempt to join/register via
 				return
 			}
 		} else {
+			if config.CurrServer != "" {
+				logger.Log(0, "WARNING: Joining any network on another server will disconnect netclient from the networks of the current server ->", config.CurrServer)
```
Comment (abhishek9686) [inline cmd/join.go:34]: use `slog`

File: functions/register.go [hunk 2]
```diff
@@ -28,6 +29,9 @@ func Register(token string, isGui bool) error {
 	if err = json.Unmarshal(data, &serverData); err != nil {
 		logger.FatalLog("could not read enrollment token")
 	}
+	if config.CurrServer != "" && config.CurrServer != strings.TrimSpace(serverData.Server[4:len(serverData.Server)]) {
+		fmt.Println("WARNING: Joining any network on another server will disconnect netclient from the networks of the current server ->", config.CurrServer)
+	}
```
Comment (abhishek9686) [inline functions/register.go:34]: you can do this after the register response, you will have the server name in the response, you can avoid slicing the data

**Review by abhishek9686 — CHANGES_REQUESTED** [review]

File: functions/register.go (hunk 2 above)
Comment (uGiFarukh) [reply functions/register.go]: done

**Review by abhishek9686 — APPROVED** [review]

**Pull request merged**